docs/develop.md

```diff
@@ -50,3 +50,42 @@ make delete-bootstrap-cluster
 ```
 
 If you have any trouble finding the right command, then you can use `make help` to get a list of all available make targets.
+
+## Toggle between local_mode and remote mode 
+We can retrieve cluster-stacks in two modes. One way is to let the controller fetch it from GitHub which is remote mode and other is we mount the cluster-stacks inside the container and start the controller with appropriate flag. 
```

Comment:
you don't have to specify the flag, remember? This is the feature of CSO, not of CSPO. Can you adapt this to not confuse people?

Reply:
But IMO we should also introduce this feature in the CSPO, right? Because when you are using `local_mode` you can hit  `GitHub API rate limit exceeded` error. In CSO, you resolved it by changing these [lines](https://github.com/SovereignCloudStack/cluster-stack-operator/pull/51/files#diff-c444f711e9191b53952edb65bfd8c644419fc7695c62611dc0fb304b4fb197d6R122-R127), am i correct?

Reply:
yes but github will never be called anyway if the cluster stack release is available in the container already. So the flag is mainly to prevent the situation that the user uses a certain release, expects this version to be available local, and in reality accidentally downloads it from a remote repository. 
By using a fake github client this can be prevented. 

It's a small thing but if you'd like to implement it, definitely go ahead! And maybe it's correct that you would call Github unnecessarily in case a release is not available locally and it always tries to find it on Github.

Reply:
So I think we will also implement this feature for CSPO because [here](https://github.com/SovereignCloudStack/cluster-stack-provider-openstack/blob/main/internal/controller/openstackclusterstackrelease_controller.go#L107-L118)(this is the same part of code, which is also in the CSO) it is called every time you spawn tilt and apply/create `openstackclusterstackrelease`. So if you do make tilt-up and then destroy the bootstrap cluster and you make tilt-up again in a short period of time then the error can come up cause of [this](https://github.com/SovereignCloudStack/cluster-stack-provider-openstack/blob/main/vendor/github.com/SovereignCloudStack/cluster-stack-operator/pkg/github/client/github_client.go#L173-L190), when you do not provide `GIT_ACCESS_TOKEN_B64` variable.

Reply:
@janiskemper 
Why do you create a gh client before checking if the assets should be downloaded? Why not create only, in the `download` condition?

Reply:
I'm not 100% sure whether I understand your use case @michal-gubricky .. But I guess if you think we should implement it, we can create an issue out of this and continue there!
@chess-knight it might be useless. If you think there is no point, then you can very well be correct! We should create it only if we need it!

Reply:
Feel free to open another issue @chess-knight

Reply:
Ok issue is here https://github.com/SovereignCloudStack/cluster-stack-provider-openstack/issues/80

@michal-gubricky then we probably do not need fake gh client in the local mode

Reply:
local mode introduces a fake github for cases where the cluster stack is not available in the container.. otherwise you won't need the local mode at all

Reply:
because the "normal" mode already does this.. Downloading only when necessary